frmts/zarr/zarr.h

```diff
@@ -812,24 +948,478 @@ class ZarrArray final : public GDALPamMDArray
+    ZarrV3CodecBlosc &operator=(const ZarrV3CodecBlosc &) = delete;
+
+  public:
+    static constexpr const char *NAME = "blosc";
+
+    ZarrV3CodecBlosc();
+    ~ZarrV3CodecBlosc() override;
+
+    IOType GetInputType() const override
+    {
+        return IOType::BYTES;
+    }
+    IOType GetOutputType() const override
+    {
+        return IOType::BYTES;
+    }
+
+    static CPLJSONObject GetConfiguration(const char *cname, int clevel,
+                                          const char *shuffle, int typesize,
+                                          int blocksize);
+
+    bool
+    InitFromConfiguration(const CPLJSONObject &configuration,
+                          const ZarrArrayMetadata &oInputArrayMetadata,
+                          ZarrArrayMetadata &oOutputArrayMetadata) override;
+
+    std::unique_ptr<ZarrV3Codec> Clone() const override;
+
+    bool Encode(const ZarrByteVectorQuickResize &abySrc,
+                ZarrByteVectorQuickResize &abyDst) const override;
+    bool Decode(const ZarrByteVectorQuickResize &abySrc,
+                ZarrByteVectorQuickResize &abyDst) const override;
+};
+
+/************************************************************************/
+/*                           ZarrV3CodecEndian                          */
+/************************************************************************/
+
+class ZarrV3CodecEndian final : public ZarrV3Codec
+{
+    bool m_bLittle = true;
+
+  public:
+    static constexpr const char *NAME = "endian";
+
+    ZarrV3CodecEndian();
+    ~ZarrV3CodecEndian() override;
+
+    IOType GetInputType() const override
+    {
+        return IOType::ARRAY;
+    }
+    IOType GetOutputType() const override
+    {
+        return IOType::BYTES;
+    }
+
+    static CPLJSONObject GetConfiguration(bool bLittle);
+
+    bool
+    InitFromConfiguration(const CPLJSONObject &configuration,
+                          const ZarrArrayMetadata &oInputArrayMetadata,
+                          ZarrArrayMetadata &oOutputArrayMetadata) override;
+
+#if CPL_IS_LSB
+    bool IsNoOp() const override
+    {
+        return m_oInputArrayMetadata.oElt.nativeSize == 1 || m_bLittle;
+    }
+#else
+    bool IsNoOp() const override
+    {
+        return m_oInputArrayMetadata.oElt.nativeSize == 1 || !m_bLittle;
+    }
+#endif
+
+    std::unique_ptr<ZarrV3Codec> Clone() const override;
+
+    bool Encode(const ZarrByteVectorQuickResize &abySrc,
+                ZarrByteVectorQuickResize &abyDst) const override;
+    bool Decode(const ZarrByteVectorQuickResize &abySrc,
+                ZarrByteVectorQuickResize &abyDst) const override;
+};
+
+/************************************************************************/
+/*                         ZarrV3CodecTranspose                         */
+/************************************************************************/
+
+class ZarrV3CodecTranspose final : public ZarrV3Codec
+{
+    // m_anOrder, such that dest_shape[i] = source_shape[order[i]]
```

Comment:
leftover?

Reply:
just a comment attempt :-)
I've clarified it to
```
    // m_anOrder is such that dest_shape[i] = source_shape[m_anOrder[i]]
    // where source_shape[] is the size of the array before the Encode() operation
    // and dest_shape[] its size after.
    // m_anOrder[] describes a bijection of [0,N-1] to [0,N-1]
```